### PR TITLE
Ommit display of blank body lines

### PR DESCRIPTION
**Pull request by puppetpies**
When developing my WIre tool i discovered in body mode that there are blank lines which are a size of 2.

I was thinking maybe implement it in library however feels wrong at the lower level.

This isn't really a pull request its whatever you make / choose todo but for my tool i don't want blank lines in my database.

Just thought i would make you aware of it !

Regards,

Brian

### CODE REVIEW

**Comment by maiha** [comment]
Hi, Brian!

> however feels wrong at the lower level

It should be I think :)

> i don't want blank lines in my database

I agree with your use case, but ignoring blank lines sometimes breaks other cases.
For example, I can image the case of monitoring ping messages that send some white spaces.

So, why don't you add a new option to control the behavior? I love the explicit way like that.
Well, candidates are ...
- "-w" : we can easily image White spaces ( `diff(1)` uses this as `ignore all white space`)

Thought?

File: examples/tcpsniffer.cr [hunk 1]
```diff
@@ -29,11 +30,15 @@ opts = OptionParser.new do |parser|
   parser.on("-b", "Body printing mode"    ) { bodymode = true }
   parser.on("-x", "Show hexdump output"   ) { hexdump  = true }
   parser.on("-v", "Show verbose output"   ) { verbose  = true }
+  parser.on("-w", "We can easily manage White spaces ( diff(1) uses this as ignore all white space)") { whitespace = true }
```
Comment (maiha) [inline examples/tcpsniffer.cr:33]: Oh, that message was a comment for you :)
Please write like this for the description.
`"Ignore all packets that contain only white spaces"`

File: examples/tcpsniffer.cr [hunk 2]
```diff
@@ -49,7 +54,13 @@ begin
     next if dataonly && !pkt.tcp_data?
 
     if bodymode
-      puts "%s: %s" % [pkt.packet_header, pkt.tcp_data.to_s.inspect]
+      if whitespace
```
Comment (maiha) [inline examples/tcpsniffer.cr]: imaged just skip the case of blank packets like this.

```
     if bodymode
+      next if whitespace && pkt.tcp_data.to_s =~ /\A\s*\Z/
       puts "%s: %s" % [pkt.packet_header, pkt.tcp_data.to_s.inspect]
```

**Review by maiha — CHANGES_REQUESTED** [review]

File: examples/tcpsniffer.cr (hunk 2 above)
Comment (puppetpies) [reply examples/tcpsniffer.cr]: I like it will change shortly !

**Comment by puppetpies** [comment]
Ok code now in place !

**Comment by puppetpies** [comment]
Also removed constant as no longer required !

**Review by maiha — APPROVED** [review]

**Comment by maiha** [comment]
Ok, it seems nice hack! I'll merge this ASAP.
Could you merge these 5 commits into 1 commit?

**Comment by puppetpies** [comment]
I will close this now open new thread !

**Comment by maiha** [comment]
FYI @puppetpies:

> I was thinking maybe implement it in library however feels wrong at the lower level.

Found this problem, and fixed the bug!
https://github.com/maiha/pcap.cr/commit/a1307f7970d183880b5f43020d22b81293210ae7

Regards,
maiha

**Pull request closed without merging**